cpu/msp430-common/lpm_cpu.c

```diff
@@ -47,16 +48,9 @@ enum lpm_mode lpm_set(enum lpm_mode target)
 
     /* ensure that interrupts are enabled before going to sleep,
        or we're bound to hang our MCU! */
-    if (target != LPM_ON) {
-        if ((__read_status_register() & GIE) == 0) {
-            printf("WARNING: entering low-power mode with interrupts disabled!\n");
-            printf("         Forcing GIE bit to 1!\n\n");
-            __bis_status_register(GIE);
-        }
-    }
+    assert((target == LPM_ON) || (__read_status_register() & GIE));
```

Comment:
Mine says `if ((target == LPM_ON) || (GIE set)) continue` (note that I don't check if the result of the bit-and is zero. That's logically the same as `if ((target != LPM_ON) && (GIE not set)) abort`. Sure I don't set the GIE bit, see https://github.com/RIOT-OS/RIOT/pull/3569#issuecomment-128086953 why I made this decision.